src/cljs/villagebookUI/components/signup.cljs

```diff
@@ -1,59 +1,44 @@
 (ns villagebookUI.components.signup
   (:require [reagent.core :as r]
-            [reagent.session :as session]
-            [accountant.core :as accountant]
+            [villagebookUI.helpers :as helpers :refer [validate-input]]
+            [villagebookUI.components.utils :refer [reqd-input]]
+            [villagebookUI.api.auth :as auth-api]))
 
-            [villagebookUI.api.auth :as auth]
-            [villagebookUI.api.user :as user-api]
-            [villagebookUI.store.user :as user-store]))
-
-(defn validate!
-  [elementID formdata]
-  (if (.checkValidity (.getElementById js/document elementID))
-    (swap! formdata assoc-in [:validation-classes elementID] "")
-    (swap! formdata assoc-in [:validation-classes elementID] "check-invalid")))
-
-(defn input [formdata k placeholder type & required]
-  [:input.form-control.mt-3.mt-3 {:placeholder placeholder
-                                  :name        k
-                                  :id          k
-                                  :on-change   #(swap! formdata assoc-in [:user k] (-> % .-target .-value))
-                                  :on-blur     #(validate! (name k) formdata)
-                                  :type        type
-                                  :required    required
-                                  :class       (str
-                                                (get-in @formdata [:validation-classes (name k)]) " "
-                                                (get-in @formdata [:validation-classes "signup-form"]))}])
+(defn submit-signup
+  [user show-validness show-error]
+  (helpers/submit-auth-form :signup-form auth-api/signup user show-validness show-error))
 
 (defn signup []
```

Comment:
I avoided making this a common component since the forms (layout, validation, error handling) are very particular to this case. I can't see it being used as it is elsewhere. Will do this if the pattern gets repeated again.

Reply:
Sounds good.